test/docker-e2e/e2e_ibc_test.go

```diff
@@ -21,13 +21,113 @@ import (
 	"github.com/celestiaorg/tastora/framework/docker/ibc"
 	"github.com/celestiaorg/tastora/framework/docker/ibc/relayer"
 	"github.com/celestiaorg/tastora/framework/testutil/wait"
-	tastoratypes "github.com/celestiaorg/tastora/framework/types"
+	sdktx "github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	icacontrollertypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/controller/types"
+	icatypes "github.com/cosmos/ibc-go/v8/modules/apps/27-interchain-accounts/types"
 	ibctransfertypes "github.com/cosmos/ibc-go/v8/modules/apps/transfer/types"
 	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	"github.com/stretchr/testify/suite"
 	"golang.org/x/sync/errgroup"
 )
 
+func TestIBCTestSuite(t *testing.T) {
+	suite.Run(t, new(IBCTestSuite))
+}
+
+// IBCTestSuite provides common IBC test infrastructure
+type IBCTestSuite struct {
+	CelestiaTestSuite
+
+	// IBC-specific fields set up in SetupTest
+	chainA          *docker.Chain // Celestia
+	chainB          *docker.Chain // Simapp
+	hermes          *relayer.Hermes
+	connection      ibc.Connection
+	transferChannel ibc.Channel
+
+	// Transaction clients
+	celestiaTxClient *user.TxClient
+}
+
+// setupIBCInfrastructure sets up IBC test infrastructure with specified app version
+func (s *IBCTestSuite) setupIBCInfrastructure(appVersion uint64) {
```

Comment:
this is the logic from the regular ibc test pulled out so it can be re-used in test setup